Publishing existing package overrides existing package to 0 bytes file

**Describe the bug**
When trying to publish a package that already exists it says "npm ERR! publish fail Cannot publish over existing version."
But at the same time, something fails at the server and the existing file turns into a 0 bytes size file.

It doesn't happen all the time, but it happens.

**To Reproduce**
Steps to reproduce the behavior:
1. Try publishing a bunch of tgz packed packages with the command "npm publish xxxx.tgz" . (that already exist). I added a file at the bottom with some tgzs and script to execute it fast 
2. get inside the server's storage place and execute command "find -iname *.tgz -type f -size -1c".
It will show you all tgz files under 1 byte.
3. Try installing the package you found and you'll get integrity error (0 bytes).

**Expected behavior**
Expected the package to not get overwritten.

**Screenshots**
Found files that are 0 bytes. 
Let's look at the `debug` package. 
![terminal](https://user-images.githubusercontent.com/1210768/51197412-1bdefd00-18fa-11e9-8521-ab04341403bc.JPG)

Found in the logs a error that appers only on package that have been destroyed. It is marked
![log](https://user-images.githubusercontent.com/1210768/51197470-4d57c880-18fa-11e9-856c-2846c28bd3e3.JPG)



**Docker || Kubernetes (please complete the following information):**
- Openshift
- Docker - verdaccio:3.10.1

**Configuration File (cat ~/.config/verdaccio/config.yaml)** 
```
# path to a directory with all packages
storage: ./storage

auth:
  htpasswd:
    file: ./htpasswd
    # Maximum amount of users allowed to register, defaults to "+inf".
    # You can set this to -1 to disable registration.
    #max_users: 1000

# a list of other known repositories we can talk to



packages:
  '@*/*':
    # scoped packages
    access: $all
    publish: $all

  '**':
    # allow all users (including non-authenticated users) to read and
    # publish all packages
    #
    # you can specify usernames/groupnames (depending on your auth plugin)
    # and three keywords: "$all", "$anonymous", "$authenticated"
    access: $all

    # allow all known users to publish packages
    # (anyone can register by default, remember?)
    publish: $all

    # if package is not available locally, proxy requests to 'npmjs' registry


# To use `npm audit` uncomment the following section
middlewares:
  audit:
    enabled: true

# log settings
logs:
  - {type: stdout, format: pretty, level: http}
  #- {type: file, path: verdaccio.log, level: info}
```

**Debugging output**
Logs at the pictures.

**Additional context**
adding a file with some packages for you to try.
[packages-demo.zip](https://github.com/verdaccio/verdaccio/files/2760664/packages-demo.zip)

1. Upload them all with the script.bat inside
2. Run the script.bat again
3. Look inside the server for 0 bytes packages.

--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
   },
   "dependencies": {
     "@verdaccio/file-locking": "0.0.8",
-    "@verdaccio/local-storage": "1.1.6",
+    "@verdaccio/local-storage": "1.1.7",
     "@verdaccio/streams": "1.0.0",
     "JSONStream": "1.3.5",
     "async": "3.0.1-0",

--- a/src/lib/local-storage.js
+++ b/src/lib/local-storage.js
@@ -222,7 +222,9 @@ class LocalStorage implements IStorage {
       metadata = cleanUpReadme(metadata);
       metadata.contributors = normalizeContributors(metadata.contributors);
 
-      if (data.versions[version] != null) {
+
+      const hasVersion = data.versions[version] != null;
+      if (hasVersion) {
         return cb( ErrorCode.getConflict() );
       }
 

--- a/test/unit/api/local-storage.spec.js
+++ b/test/unit/api/local-storage.spec.js
@@ -364,16 +364,10 @@ describe('LocalStorage', () => {
         test('should fails on add a duplicated new tarball ', (done) => {
           const tarballData = JSON.parse(readMetadata('addTarball'));
           const stream = storage.addTarball(pkgName, tarballName);
-          let spy;
-          // $FlowFixMe
-          spy = jest.spyOn(stream &&  stream._readableState && stream._readableState.pipes, 'abort');
           stream.on('error', (err) => {
             expect(err).not.toBeNull();
             expect(err.statusCode).toEqual(HTTP_STATUS.CONFLICT);
             expect(err.message).toMatch(/this package is already present/);
-          });
-          stream.on('success', function(){
-            expect(spy).toHaveBeenCalled();
             done();
           });
           stream.end(new Buffer(tarballData.data, 'base64'));

